src/js/tabs/firmware_flasher.js

```diff
@@ -264,44 +276,58 @@ TABS.firmware_flasher.initialize = function (callback) {
+                    items[`${boardName}-${manufacturerId}`] = { isUnifiedTarget: true, boardName: boardName, manufacturerId: manufacturerId, targetName: targetName };
+                } else {
+                    items[boardName] = { isUnifiedTarget: true, targetName: targetName };
+                }
             });
             Object.keys(builds).forEach(function (key) {
-                // releases is under the hood, so we can have duplicate entries
-                var legacyKey = key + " (Legacy)";
-                if (unifiedConfigs[key] === undefined) {
-                    items[key] = "something";
-                    releases[key] = builds[key];
+                releases[key] = builds[key];
+
+                let legacyKey = key + "_legacy";
+                if (!unifiedTargetNames.includes(key)) {
+                    items[key] = { isUnifiedTarget: false, targetName: key };
                 } else {
-                    items[legacyKey] = "i18nplz";
-                    baseTargets[legacyKey] = key;
-                    releases[legacyKey] = builds[key];
-                    releases[key] = builds[key];
+                    items[legacyKey] = { isUnifiedTarget: false, boardName: key, targetName: key };
                 }
             });
-            $('select[name="board"]').empty()
+            var boards_e = $('select[name="board"]');
+            var versions_e = $('select[name="firmware_version"]');
+            boards_e.empty()
                 .append($("<option value='0' i18n='firmwareFlasherOptionLabelSelectBoard'></option>"));
 
-            $('select[name="firmware_version"]').empty()
+            versions_e.empty()
                 .append($("<option value='0' i18n='firmwareFlasherOptionLabelSelectFirmwareVersion'></option>"));
-            var boards_e = $('select[name="board"]');
-            var versions_e = $('select[name="firmware_version"]');
             var selectTargets = [];
             Object.keys(items)
                 .sort()
                 .forEach(function(target, i) {
-                    var select_e = $("<option value='{0}'>{1}</option>".format(target,
-                        items[target] === "i18nplz" ? i18n.getMessage("firmwareFlasherLegacyLabel",
-                        {target: baseTargets[target]}) : target));
+                    let item = items[target];
+                    let optionText;
+                    if (item.boardName) {
+                        if (!item.isUnifiedTarget) {
+                            optionText = i18n.getMessage("firmwareFlasherLegacyLabel", { target: items[target].boardName });
+                        } else {
+                            optionText = `${item.boardName} (${item.manufacturerId})`;
+                        }
+                    } else {
+                        optionText = target;
+                    }
+
+                    var select_e = $("<option value='{0}'>{1}</option>".format(item.targetName, optionText));
```

Comment:
I don't get legacy targets listed when I run this PR
![image](https://user-images.githubusercontent.com/190571/65726488-0f569b00-e06a-11e9-8107-26bbfc36b881.png)
![image](https://user-images.githubusercontent.com/190571/65726031-eda8e400-e068-11e9-9ea3-39177386dafa.png)

Reply:
Ha, I was wondering why we were adding the releases list twice - now this makes sense. Will check to see how we can make this a bit more intuitive.

Reply:
Ok, some more rewriting needed to make this work with the added complexity. Done now, please test again.